wcfsetup/install/files/lib/system/template/TemplateScriptingCompiler.class.php

```diff
@@ -1181,8 +1244,26 @@ public function compileVariableTag($tag, $replaceQuotes = true) {
 						if (/*strpos($values[$i], '$') !== false || */strpos($values[$i], '@@') !== false) {
 							throw new SystemException(static::formatSyntaxError("unexpected '->".$values[$i]."' in tag '".$tag."'", $this->currentIdentifier, $this->currentLineNo));
 						}
-						if (strpos($values[$i], '$') !== false) $result .= '{'.$this->compileSimpleVariable($values[$i], $variableType).'}';
-						else $result .= $values[$i];
+						if (strpos($values[$i], '$') !== false) {
+							$result .= '{'.$this->compileSimpleVariable($values[$i], $variableType).'}';
+						}
+						else {
+							$result .= $values[$i];
+						}
+						
+						// `WCF::getDB()` and `WCF::getTPL()` are not supported in enterprise mode
```

Comment:
It is very easy to bypass this security feature by using the following code: 

```
{assign var='wcf' value=$__wcf}
{$wcf->getDB()|wcfDebug}
```

So you may disable assigning `$__wcf` to other variables.

Reply:
Also the following is possible: 

```
{$__wcf->__get('DB')|wcfDebug}
```

Reply:
```
{$__wcf->__call('getDB', $__wcf->getUser()->getData())|wcfDebug}
```

Reply:
We can resolve these by using a `EnterpriseModeWCF` class that extends the `WCF` class and is used in the template. It will restrict the access, while mapping properties and methods to the original object:
```php
class EnterpriseModeWCF extends WCF {
  private $wcf;

  public function __construct(WCF $wcf) {
    $this->wcf = $wcf;
  }

  // magic
}
```